src/actions/TemperatureSyncActions.js

```diff
@@ -85,9 +103,145 @@ const downloadLogs = sensor => async dispatch => {
     dispatch(downloadLogsError());
   }
 
+  return new Promise(resolver => resolver(success));
+};
+
+const startResettingAdvertisementFrequency = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.START_RESETTING_ADVERTISEMENT_FREQUENCY,
+});
+const completeResettingAdvertisementFrequency = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.COMPLETE_RESETTING_ADVERTISEMENT_FREQUENCY,
+});
+const errorResettingAdvertisementFrequency = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.ERROR_RESETTING_ADVERTISEMENT_FREQUENCY,
+});
+
+const resetAdvertisementFrequency = sensor => async dispatch => {
+  dispatch(startResettingAdvertisementFrequency());
+
+  const resettingAdvertisementFrequencyResult = await sensor.resetAdvertisementFrequency();
+
+  const { success } = resettingAdvertisementFrequencyResult;
+
+  if (success) dispatch(completeResettingAdvertisementFrequency());
+  else dispatch(errorResettingAdvertisementFrequency());
+
+  return new Promise(resolver => resolver(success));
+};
+
+const startResettingLogFrequency = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.START_RESETTING_LOG_FREQUENCY,
+});
+const completeResettingLogFrequency = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.COMPLETE_RESETTING_LOG_FREQUENCY,
+});
+const errorResettingLogFrequency = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.ERROR_RESETTING_LOG_FREQUENCY,
+});
+
+const resetLogFrequency = sensor => async dispatch => {
+  dispatch(startResettingLogFrequency());
+
+  const resettingLogFrequencyResult = await sensor.resetLogFrequency();
+
+  const { success } = resettingLogFrequencyResult;
+
+  if (success) dispatch(completeResettingLogFrequency());
+  else dispatch(errorResettingLogFrequency());
+
+  return new Promise(resolver => resolver(success));
+};
+
+const startSavingTemperatureLogs = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.START_SAVING_TEMPERATURE_LOGS,
+});
+const completeSavingTemperatureLogs = () => ({
+  type: TEMPERATURE_SYNC_ACTIONS.COMPLETE_SAVING_TEMPERATURE_LOGS,
+});
+
+const createTemperatureLogs = sensor => dispatch => {
+  const { sensorLogs, location } = sensor;
+
+  // Only create TemperatureLogs for the greatest multiple of 6 SensorLogs,
+  // as each SensorLog is a 5 minute log, and each Temperature log a 30 minute log.
+  const iterateToo = sensorLogs.length - (sensorLogs.length % 6);
+  const sensorLogsToGroup = sensorLogs.sorted('timestamp').slice(0, iterateToo);
+  const groupedSensorLogs = chunk(sensorLogsToGroup, 6);
+
+  dispatch(startSavingTemperatureLogs(groupedSensorLogs.length));
+
+  UIDatabase.write(() => {
+    groupedSensorLogs.forEach(sensorLogGroup => {
+      const { isInBreach, mostRecentTemperatureBreach } = sensor;
+      const temperature = Math.min(...sensorLogGroup);
+      const timestamp = Math.min(...sensorLogGroup);
```

Comment:
These two lines need to have `sensorLogGroup.map(({timestamp}) => timestamp)` - fixed in #2650